src/linear_algebra/matrix.lean

```diff
@@ -721,7 +727,7 @@ calc  matrix.trace ι R R (linear_map.to_matrix hb hb f)
 open_locale classical
 
 theorem trace_aux_range (R : Type u) [comm_ring R] {M : Type v} [add_comm_group M] [module R M]
-  {ι : Type w} [fintype ι] {b : ι → M} (hb : is_basis R b) :
+  {ι : Type w} [decidable_eq ι] [fintype ι] {b : ι → M} (hb : is_basis R b) :
```

Comment:
Why is this decidability assumption now necessary?

Reply:
It should not be harder to apply the function: `trace_aux` already takes a `decidable_eq ι` argument, so it should be found by unification.